analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UnchangedLocalVariablesShouldBeConst.cs

```diff
@@ -463,4 +463,14 @@ public void IndexerIndexedByLambda_CrazyCoverage()
         }
         set { }
     }
+
+    public void Message()
+    {
+        var s1 = "Test";                              // Noncompliant {{Add the 'const' modifier to 's1', and replace 'var' with 'string'.}}
+        string s2 = $"This is a {nameof(Message)}";   // Noncompliant {{Add the 'const' modifier to 's2', and use string concatenation instead of an interpolated string.}}
+        var s3 = $"This is a {nameof(Message)}";      // Noncompliant {{Add the 'const' modifier to 's3', and replace 'var' with 'string', and use string concatenation instead of an interpolated string.}}
```

Comment:
> A better error message would be:
Add the 'const' modifier to 'message', replace 'var' with 'string', and use string concatenation instead of an interpolated string.

I don't agree with this case. What about
> $"This is a {const1} {const2} {const3} {const4} ... {const10}";

do we really want to tell users to use string concatenation?

IMO we shouldn't raise an issue in this case, to avoid noise

Reply:
The scope for interpolated string constants is fairly limited:

> Interpolated strings ${}, provided that all components are constant expressions of type string and all interpolated components lack alignment and format specifiers.

[Source](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/constant_interpolated_strings)

There are only two use cases for this AFAIK:

1. Building a constant string out of other constant strings
2. `nameof`

In your example, `const1` .. `const10` must all be string constants to raise the rule.
According to the spec, it is always possible to use the contact operator for any interpolated string, that can be turned into a const:

> The current + operator for string concatnation can combine strings in a similar manner to the current proposal.

Therefore the suggestion is always feasible.

I think these options are valid:

1. Raise and provide guidance on how to turn it into a valid const (current behavior)
2. Don't raise for verbatim interpolated strings for projects lower than C# 10.
3. Don't raise any interpolated string for projects lower than C#10.

What do you think?

Reply:
> and use string concatenation instead of an interpolated string.

> In your example, const1 .. const10 must all be string constants to raise the rule.

Sure, my main question is why would

> const string s3 = "This is a " + nameof(Message) + " some other text " + otherConst; // and another 4 concatenations

be more readable than using an interpolated string?

the `" + "` makes code less readable IMO

I would say not raise the issue when only concatenation is the answer to avoid subjective interpretation (so yes, don't raise when less than C# 10).

Reply:
An example of subjective interpretation: "I find string interpolation more readable than concatenation, so I don't want to use `const string` in this case".

I am open to other ideas though, if you think this won't spark subjective reactions from devs.

Reply:
I also prefer to not raise the issue. I just wanted to make sure, you have the full context. Let me change the implementation.

Reply:
The context is: we shouldn't raise issues which are too controversial (noise) especially for SonarWay rules.